.github/workflows/tox-pytest.yml

```diff
@@ -0,0 +1,93 @@
+name: tox-pytest
+
+on: [ push, pull_request ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+      fail-fast: false
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+    - uses: actions/checkout@v2.4.0
+      with:
+        fetch-depth: 2
+
+    - name: Set up conda environment for testing
+      uses: conda-incubator/setup-miniconda@v2.1.1
+      with:
+        mamba-version: "*"
+        channels: conda-forge,defaults
+        channel-priority: true
+        python-version: ${{ matrix.python-version }}
+        activate-environment: pudl-rmi
+        environment-file: environment.yml
+
+    - name: Log the conda environment
+      run: |
+        conda info
+        conda list
+        conda config --show-sources
+        conda config --show
+        printenv | sort
+
+    - name: Cache PUDL DB
+      uses: actions/cache@v2
+      with:
+        path: ~/pudl-work/sqlite/pudl.sqlite
+        # The last time we deployed Datasette. Update if we redeploy.
+        key: "2021-11-12"
+
+    - name: Download PUDL DB and log pre-test PUDL workspace contents
+      run: |
+        mkdir -p ~/pudl-work/sqlite/
+        if [ ! -f ~/pudl-work/sqlite/pudl.sqlite ]
+        then curl -o ~/pudl-work/sqlite/pudl.sqlite https://data.catalyst.coop/pudl.db
+        fi
+        find ~/pudl-work/
```

Comment:
where/when has the pudl-work directory been setup in this process??

Reply:
The directory is created a couple of lines above, and we aren't running `pudl_setup()` anywhere because we only need the PUDL DB and aren't going to be doing any ETL.